### ISSUE TITLE
Smart alerts don't respect selected calendars

### ISSUE THREAD

**Issue by daividssilverio** [toggl/mobileapp#4652]
## 🐛 Bug Report
I received one smart alert notification from an event that didn't make part of the calendars selected to be integrated with the app through the app settings.

## Context
Found it while doing tests for #4601

Build/Branch: `project/calendroid`
Phone OS: Android
Phone model: S8

## Expected behavior
Only events from the calendars selected in the calendar settings should receive smart alerts at the times set in the calendar smart alerts settings.

## To reproduce
Make sure you have at least 2 calendars in your system (`A`, `B`).
1. Create two events, `a` and `b`. (`a` in the `A` calendar, `b` in the `B` calendar)
2. Enable the calendar integration in the app, selecting both calendars
3. Make sure you see the events in the calendar
4. Set your calendar smart alerts settings such that you can receive the events in reasonable time (or find some adb magic, up to you).
5. Now go to the calendar settings, de-select the `B` calendar.
6. Go back to the settings page.
7. Don't go back to the calendar page.
8. Close the app, leave it be, just wait for time of your smart alerts
9. You will receive the notifications for both events `a` and `b`.

## Suggestions
It seems this is just a matter of updating the scheduled events correctly when the calendars integrated with the app changes. We have to unschedule previously scheduled events from calendars that are not integrated anymore.

**Comment by juanlaube** [comment]
@daividssilverio I can't reproduce this, in fact, I've been receiving events only from the calendar I selected since I started testing last week. Can you try again and confirm this is a problem?

**Comment by daividssilverio** [comment]
Yup, I can still replicate it.
And worse, I found something that seems to be bug as well.

After de-selecting one of the calendars, I received still both events.
Then I tried changing the smart reminder settings by coincidence and when I changed it to `When events starts`, both notifications were fired 😱 .

**Comment by daividssilverio** [comment]
@juanlaube I'm able to replicate it and to diagnose the problem.
First, that's how we are cancelling the scheduled alarms in android:
```
 var context = Application.Context;
                var notificationManager = NotificationManagerCompat.From(context);
                notificationManager.CancelAll();
```
☝️ That cancels notifications, not pending intents scheduled by the AlarmManager.
That's most likely the source of our problems.

Second, you can check that the schedules are wrong using
```
adb shell dump sys alarm | grep SmartAlertCalendarEventBroadcastReceiver | wc -l
```
☝️ that will give you the number of pending intents for that broadcast received. It goes up and up as you play with the selected calendars.

I managed to reduce the problem adding data to the original intents so that we can find the pending intents and cancel them, but the problem escalates and I couldn't fix it when I checked what happened when changing the "Smart reminders".

Anyway, this looks very fixable, I'll take another try soon.

**Comment by daividssilverio** [comment]
Closed by #4710
